noetic/ros-opencog/Dockerfile

```diff
@@ -0,0 +1,167 @@
+     echo "fa_IR.UTF-8 UTF-8" >> /etc/locale.gen && \
+     echo "ar_AE.UTF-8 UTF-8" >> /etc/locale.gen && \
+     echo "kk_KZ.UTF-8 UTF-8" >> /etc/locale.gen && \
+     echo "tr_TR.UTF-8 UTF-8" >> /etc/locale.gen)
+
+# WTF. In debian wheezy, it is enough to just say locale-gen without
+# any arguments. But in trusty, we eneed to be explicit.  I'm confused.
+# RUN locale-gen
+# Note also: under trusty, fa_IR.UTF-8 causes locale-gen to fail,
+# must use the naked  fa_IR
+# Note also: Kazakh is kk_KZ not kz_KZ
+RUN locale-gen en_US.UTF-8 ru_RU.UTF-8 he_IL.UTF-8 \
+     de_DE.UTF-8 lt_LT.UTF-8 fa_IR ar_AE.UTF-8 kk_KZ.UTF-8 tr_TR.UTF-8
+
+# Build the libraries and command-line parser only
+# Assumes that the sources have already been unpacked.
+RUN (cd link-grammar-5*; mkdir build; cd build; \
+     ../configure; make -j12; make install; ldconfig)
+
+RUN adduser --disabled-password --gecos "Link Parser User" link-parser
+
+USER link-parser
+RUN (cd /home/link-parser; echo "export LANG=en_US.UTF-8" >> .bash_aliases)
+CMD bash
+
+RUN export LANG=en_US.UTF-8
+
+# -------------------------------------------------------------
+USER root
+WORKDIR /opencog
+
+# Change line below on rebuild. Will use Docker cache up to this line,
+# twigging this date will update the opencog sources.
+ENV LAST_SOFTWARE_UPDATE 2016-03_13
+
+# Git pull for all packages
+RUN cd /opencog/ && find . -maxdepth 1 -mindepth 1 -type d \
+	-execdir git --git-dir=$PWD/{}/.git --work-tree=$PWD/{} pull \;
+
+# Build the base software.
+RUN (mkdir /opencog/cogutil/build; cd /opencog/cogutil/build; \
+	cmake ..; make -j6; make install)
+RUN (mkdir /opencog/atomspace/build; cd /opencog/atomspace/build; \
+	cmake ..; make -j12; make install)
+RUN (mkdir /opencog/lg-atomese/build; cd /opencog/lg-atomese/build; \
+    cmake ..; make -j12; make install)
+RUN (mkdir /opencog/opencog/build; cd /opencog/opencog/build; \
+	cmake ..; make -j12; make install)
+COPY /scripts/install-ubuntu-relex.sh /opencog/relex/install-scripts/
+RUN (cd /opencog/relex; install-scripts/install-ubuntu-relex.sh; ant build; ant install)
+
+# Build the unit tests. This is .. optional, but we do it anyway,
+# for right now, just to sanity-check things.
+# WORKDIR /opencog/cogutil/build
+# RUN bash -l -c "make -j6 tests"
+#
+# WORKDIR /opencog/atomspace/build
+# RUN bash -l -c "make -j12 tests"
+#
+# WORKDIR /opencog/opencog/build
+# RUN bash -l -c "make -j12 tests"
+#
+COPY /scripts/.guile /root/
+COPY /scripts/unit-test.sh /root/
+COPY /scripts/tmux.sh /root/
+
+# Pre-compile the guile modules. We could say `guild comple ...`, but
+# this is easier. This avoids fly compilation during container start.
+RUN (bash -l -c "echo \"(use-modules (opencog))\" | guile ; \
+  echo \"(use-modules (opencog exec))\" | guile ; \
+  echo \"(use-modules (opencog logger))\" | guile ; \
+  echo \"(use-modules (opencog persist))\" | guile ; \
+  echo \"(use-modules (opencog persist-sql))\" | guile ; \
```

Comment:
replace this by `(opencog persist-rocks)`  I do not believe that any of the docker containers use postgres, anyway.  Rocks is about 3x faster, and just easier to deal with and to configure.

Reply:
which also means that postgres does not need to be installed in the base container .... but rocksDB does need to be.